### PR TITLE
Apply curie upgrade

### PR DESCRIPTION
**Pull request by chengwenxi**
### Description

[_PR description_]

### Issue Link

[_link issue here_]

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Contents

- [_item_]

### Rationale

[_design decisions and extended information_]

### How Has This Been Tested?

[_explanation_]

<hr>

## How to fill a PR description 

Please give a concise description of your PR.

The target readers could be future developers, reviewers, and auditors. By reading your description, they should easily understand the changes proposed in this pull request.

MUST: Reference the issue to resolve

### Single responsability

Is RECOMMENDED to create single responsibility commits, but not mandatory.

Anyway, you MUST enumerate the changes in a unitary way, e.g.

```
This PR contains:
- Cleanup of xxxx, yyyy
- Changed xxxx to yyyy in order to bla bla
- Added xxxx function to ...
- Refactored ....
```

### Design choices

RECOMMENDED to:
- What types of design choices did you face?
- What decisions you have made?
- Any valuable information that could help reviewers to think critically

### CODE REVIEW

**Review by anylots — APPROVED** [review]

File: zkevm-circuits/src/evm_circuit/util/common_gadget/curie.rs [hunk 1]
```diff
@@ -41,18 +41,18 @@ impl<F: Field> CurieGadget<F> {
         );
 
         // TODO: refactor
-        // is_scoll_chain means (chain_id - 534352) * (chain_id - 222222) == 0
-        let is_scroll_chain = IsZeroGadget::construct(
+        // is_morph_chain means (chain_id - 53077) * (chain_id - 2710) == 0
```
Comment (Voidkai) [inline zkevm-circuits/src/evm_circuit/util/common_gadget/curie.rs]: Wrong MAINNET_CHAIN_ID in comment, -> is_morph_chain means (chain_id - 2818)*(chain_id - 53077)== 0

**Review by Voidkai — COMMENTED** [review]
Typo in comment

File: zkevm-circuits/src/evm_circuit/util/common_gadget/curie.rs (hunk 1 above)
Comment (chengwenxi) [reply zkevm-circuits/src/evm_circuit/util/common_gadget/curie.rs]: it should be `(chain_id - 2818)*(chain_id - 53077)*(chain_id-2710) == 0`

Comment (Voidkai) [reply zkevm-circuits/src/evm_circuit/util/common_gadget/curie.rs]: But you only check the chain_id equals to  MORPH_MAINNET_CHAIN_ID or MORPH_DEVNET_CHAIN_ID, without checking the TESTNET CHAIN ID.

File: zkevm-circuits/src/evm_circuit/util/common_gadget/curie.rs [hunk 2]
```diff
@@ -41,18 +41,18 @@ impl<F: Field> CurieGadget<F> {
...
+        let is_morph_chain = IsZeroGadget::construct(
```
Comment (Voidkai) [inline zkevm-circuits/src/evm_circuit/util/common_gadget/curie.rs]: Maybe you can try https://github.com/scroll-tech/zkevm-circuits/blob/84b9bb5618f7041f0ff364a89d305ded2c0ab865/zkevm-circuits/src/evm_circuit/util/math_gadget/batched_is_zero.rs#L19, for batch constraint chain id.

**Review by Voidkai — COMMENTED** [review]
batch constraint

**Review by Voidkai — APPROVED** [review]

**Pull request merged**